### ISSUE TITLE
Support recursive unpack

### ISSUE THREAD

**Issue by ghost** [socs-dev-env/dosocs2#43]
A directory or archive may have other archives (tar, zip, jar) in it; we would like to unpack these during scanning, rather than treat them as monolithic files.

**Comment by ghost** [comment]
Sort of fixed... added the `nomos_deep` scanner which will unpack the input file if it is an archive, and return scanner results as if it were a monolithic file, like this:

``` python
[ScannerResult('archive.tar', 'Apache-2.0'),
 ScannerResult('archive.tar', 'MIT'),
 ScannerResult('archive.tar', 'Apache-possibility')]
```

It does not do recursion, so, for example, a jar inside of a tar inside of a tar will not be unpacked.

**Comment by ghost** [comment]
Merged into master with #45
